assets/js/custom/Translation.js

```diff
@@ -1,42 +1,22 @@
 /* eslint-env jquery */
 
 // eslint-disable-next-line no-unused-vars
-function Translation (translatedByLine) {
-  const providerMap = {
-    itranslate: 'iTranslate'
-  }
-  let displayLanguageMap = {}
-  let translatedByLineMap = {}
-  let targetLanguage = null
-
-  $(document).ready(function () {
-    setTargetLanguage()
-    getDisplayLanguageMap()
-    splitTranslatedByLine()
-  })
-
-  $(document).on('click', '.comment-translation-button', function () {
-    const commentId = $(this).attr('id').substring('comment-translation-button-'.length)
-
-    $(this).hide()
-
-    if (isTranslationNotAvailable(commentId)) {
-      $('#comment-translation-loading-spinner-' + commentId).show()
-      translateComment(commentId)
-    } else {
-      openTranslatedComment(commentId)
+class Translation {

```

Comment:
To help split up the program translation and comment translation functionality, the Translate function has now been replaced with a Translation base class. The new Translation class contains the same functionality as the previous Translation function by getting and splitting up the translatedByLine, then also retrieving all of the display languages to show in the translatedByLine. It also contains a new method to set the translatedByLine for both the program and comment section. However, the functions that set, opened, and removed the translations from the comment section have now been moved to the TranlsateComment class which extends this new Translation class.